cpp/open3d/core/linalg/InverseCUDA.cpp

```diff
@@ -41,31 +42,30 @@ void InverseCUDA(void* A_data,
 
     DISPATCH_LINALG_DTYPE_TO_TEMPLATE(dtype, [&]() {
         int len;
-        int* dinfo =
-                static_cast<int*>(MemoryManager::Malloc(sizeof(int), device));
+        Blob dinfo(sizeof(int), device);
 
         OPEN3D_CUSOLVER_CHECK(
                 getrf_cuda_buffersize<scalar_t>(handle, n, n, n, &len),
                 "getrf_buffersize failed in InverseCUDA");
-        void* workspace = MemoryManager::Malloc(len * sizeof(scalar_t), device);
+        Blob workspace(len * sizeof(scalar_t), device);
 
         OPEN3D_CUSOLVER_CHECK_WITH_DINFO(
-                getrf_cuda<scalar_t>(handle, n, n,
-                                     static_cast<scalar_t*>(A_data), n,
-                                     static_cast<scalar_t*>(workspace),
-                                     static_cast<int*>(ipiv_data), dinfo),
-                "getrf failed in InverseCUDA", dinfo, device);
+                getrf_cuda<scalar_t>(
+                        handle, n, n, static_cast<scalar_t*>(A_data), n,
+                        static_cast<scalar_t*>(workspace.GetDataPtr()),
+                        static_cast<int*>(ipiv_data),
+                        static_cast<int*>(dinfo.GetDataPtr())),
+                "getrf failed in InverseCUDA",
+                static_cast<int*>(dinfo.GetDataPtr()), device);
 
         OPEN3D_CUSOLVER_CHECK_WITH_DINFO(
                 getrs_cuda<scalar_t>(handle, CUBLAS_OP_N, n, n,
                                      static_cast<scalar_t*>(A_data), n,
                                      static_cast<int*>(ipiv_data),
                                      static_cast<scalar_t*>(output_data), n,
-                                     dinfo),
-                "getrs failed in InverseCUDA", dinfo, device);
-
-        MemoryManager::Free(workspace, device);
```

Comment:
We do have `Free` here. So is this a memory leak, or are you trying to be more concise by using `Blob`?

Reply:
There can actually be a leak if an exception is thrown between the allocation and the call to `Free`.
This does happen in a few unit tests, for example [here](https://github.com/intel-isl/Open3D/blob/5919ab5397cdbdf5386b097b2cfa20e75f8f4fa0/cpp/tests/core/Linalg.cpp#L358)

Reply:
:+1: